src/lib/storage/table.cpp

```diff
@@ -22,6 +22,36 @@
 #include "utils/assert.hpp"
 #include "value_segment.hpp"
 
+namespace {
+
+using namespace hyrise;  // NOLINT(build/namespaces)
+
+// Checks if the two vectors have common elements, e.g., lhs = [0, 2] and rhs = [2, 1, 3]. We expect very short
+// vectors, so we use a simple but quadratic solution. For larger vectors, we could create sets and check set
+// containment.
+bool columns_intersect(const std::vector<ColumnID>& lhs, const std::vector<ColumnID>& rhs) {
+  for (const auto column_id : lhs) {
+    if (std::find(rhs.cbegin(), rhs.cend(), column_id) != rhs.cend()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// Checks if @param lhs is a sublist of @param rhs, e.g., lhs = [0, 2, 1] and rhs = [0, 2, 1, 3, 4], or vice versa.
```

Comment:
Note which function assumes both inputs to be ordered acordingly?? `intersect` does apparently not, `overlap` does.

Besides that,  I would have expected `[0,1,2]` to overlap with `[1,2]`.

Reply:
Yeah, the point of `overlap` is to match from the beginning/have the same sequence from the start on. `columns_share_start_sequence`?

Reply:
(removed it entirely)